### PR TITLE
Support the pyproject.toml.

### PR DESCRIPTION
**Pull request by teweitsai**
Support the *pyproject.toml* file.

### CODE REVIEW

File: conda/meta.yaml [hunk 1]
```diff
@@ -8,6 +8,8 @@ source:
 
 build:
   script: python -m pip install --no-deps --ignore-installed .
+  entry_points:
+    - run_mtm2 = lsst.ts.m2.cli:run_mtm2
```
Comment (wvreeven) [inline conda/meta.yaml:12]: The `run_mtm2` function is in the `csc` module and not the `cli` module so please correct this line.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+classifiers = [ "Programming Language :: Python :: 3" ]
+urls = {documentation = "https://jira.lsstcorp.org/secure/Dashboard.jspa", repository = "https://github.com/lsst-ts/ts_m2" }
+dynamic = [ "version" ]
+
+[tool.setuptools.dynamic]
+version = { attr = "setuptools_scm.get_version" }
+
+[tool.setuptools.packages.find]
+where = [ "python" ]
+
```
Comment (wvreeven) [inline pyproject.toml:20]: Please add the entry_points here as well.

File: setup.py [hunk 3]
```diff
@@ -1,38 +1,8 @@
...
-    tests_require=tests_require,
-    extras_require={"dev": dev_requires},
-    license="GPL",
-    project_urls={
-        "Bug Tracker": "https://jira.lsstcorp.org/secure/Dashboard.jspa",
-        "Source Code": "https://github.com/lsst-ts/ts_m2",
-    },
+setup(
+    version=setuptools_scm.get_version(),
+    scripts=[str(path) for path in pathlib.Path("bin").glob("*.py")],
```
Comment (wvreeven) [inline setup.py:7]: Please remove this line and the import for pathlib as well.

**Review by wvreeven — CHANGES_REQUESTED** [review]
This is a great start but there are a few mistakes. Please correct them and request my review again.

File: setup.py (hunk 3 above)
Comment (teweitsai) [reply setup.py]: Fixed in the commit: [dc78758](https://github.com/lsst-ts/ts_m2/pull/25/commits/dc78758f743d2867d90bbb8500248fe6606a8870). Thanks!

File: pyproject.toml (hunk 2 above)
Comment (teweitsai) [reply pyproject.toml:20]: Added in the commit: [dc78758](https://github.com/lsst-ts/ts_m2/pull/25/commits/dc78758f743d2867d90bbb8500248fe6606a8870). Thanks!

File: conda/meta.yaml (hunk 1 above)
Comment (teweitsai) [reply conda/meta.yaml]: Fixed in the commit: [dc78758](https://github.com/lsst-ts/ts_m2/pull/25/commits/dc78758f743d2867d90bbb8500248fe6606a8870). Thanks!

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+write_to = "python/lsst/ts/m2/version.py"
+write_to_template = """
+# Generated by setuptools_scm
+__all__ = ["__version__"]
+__version__ = "{version}"
+"""
+
+[tool.pytest.ini_options]
+addopts = "--black --flake8 --ignore-glob=*/version.py"
+flake8-ignore = ["E133", "E203", "E226", "E228", "N802", "N803", "N806", "N812", "N813", "N815", "N816", "W503", "doc/conf.py ALL", "__init__.py ALL", "version.py ALL", "tests/.tests ALL"]
```
Comment (wvreeven) [inline pyproject.toml:34]: Note that in the latest version of the template in the confluence page, this line was updated as well to be much shorter. I'll leave it up to you to change it or not. Like this it is fine but it contains some superfluous entries that may be removed if you want.

**Review by wvreeven — APPROVED** [review]
Looks good to me!

File: pyproject.toml (hunk 4 above)
Comment (teweitsai) [reply pyproject.toml]: Updated in the commit: [835faac](https://github.com/lsst-ts/ts_m2/pull/25/commits/835faacd505ba3eb57001036a7757269e1fd4219). Thanks!

**Pull request merged**